### ISSUE TITLE
[htdf2.0] add `initial_height` in genesis file , start new chain in block initial_height

### ISSUE THREAD

**Issue by youngqqcn** [orientwalt/htdf#55]
reference:

https://github.com/cosmos/cosmos-sdk/pull/7089
https://github.com/tendermint/tendermint/pull/5191

**Comment by deep2chain** [comment]
this function was enabled after commit: https://github.com/orientwalt/htdf/commit/dedab754cc8b5dddbaf3dbed4e8cddd54f64044c , and waiting for test

- usage
```
$ hsd init mynode --chain-id $(CHAIN_ID) --initial-height 10
```

**Comment by youngqqcn** [comment]
I tested the fiture just now, it seems exists some problems.

```
$hscli query account htdf1gx4rl89cex9nzpazs75778ys5cpellssnpnsnl

ERROR: failed to load state at height 82; version does not exist (latest height: 82): invalid request


$ hscli tx send $(hscli  accounts list | head -1) htdf1xwpsq6yqx0zy6grygy7s395e2646wggufqndml  1000000000000000satoshi --gas-price=100
GetCmdSend:txBldr.GasWanted() 200000
GetCmdSend:txBldr.GasPrices(): 100
ERROR: failed to load state at height 97; version does not exist (latest height: 97): invalid request
```

**Comment by deep2chain** [comment]
you can test commit: c067bde73c128fa44621c8954f60cd82c318ff1a

**Comment by youngqqcn** [comment]
```
$ hscli query account htdf1h8g6wevgcm4r45227c3myudfdsmw4vyyfp8sjd
ERROR: failed to load state at height 149; version does not exist st.tree.Version()140 != version150 (latest height: 149): invalid request
```

**Comment by deep2chain** [comment]
check commit: b3aa428923d4a4c2ec82fb0b5f7f5c52e41df145

**Comment by youngqqcn** [comment]
When restart `hsd` it must starts with option `--initial-height   initial_height`, but `initial_height` already be setted at `genesis.json`.  


Users needn't know about this special options when they deploys a node, they just need the `genesis.json` and `config.toml`. In order to 
Could we remove this  redundant option?

**Comment by youngqqcn** [comment]
@deep2chain

**Comment by deep2chain** [comment]
@youngqqcn  your suggestion is reasonable. I think to save in DB when initializing chain and to refer it is better than using config.toml or genesis.json. Now the solution is in development.

**Comment by gaojunying** [comment]
this issue was resolved after commit: 097b2efd930335147e4454fbeb1a6a04d04f10e2

**Comment by youngqqcn** [comment]
> this issue was resolved after commit: [097b2ef](https://github.com/orientwalt/htdf/commit/097b2efd930335147e4454fbeb1a6a04d04f10e2)

basic  send test cases was failed:

```
hscli tx send $(hscli  accounts list | head -1) htdf1xwpsq6yqx0zy6grygy7s395e2646wggufqndml  1000000000000000satoshi --gas-price=100
GetCmdSend:txBldr.GasWanted() 200000
GetCmdSend:txBldr.GasPrices(): 100
ERROR: failed to load state at height 102; version does not exist st.tree.Version()2 != version93 (latest height: 102): invalid request
make: [Makefile:16：sendhtdf] 错误 1 (已忽略）
```

**Comment by youngqqcn** [comment]
https://github.com/orientwalt/htdf/commit/097b2efd930335147e4454fbeb1a6a04d04f10e2#diff-d50d50328db8879587ae3d317cc8d4020c86b5cb0e32f2a68b3a4e5d1e5338dfR191

https://github.com/orientwalt/htdf/blob/097b2efd930335147e4454fbeb1a6a04d04f10e2/app/v0/protocol_v0.go#L191

**Comment by youngqqcn** [comment]
> [097b2ef#diff-d50d50328db8879587ae3d317cc8d4020c86b5cb0e32f2a68b3a4e5d1e5338dfR191](https://github.com/orientwalt/htdf/commit/097b2efd930335147e4454fbeb1a6a04d04f10e2#diff-d50d50328db8879587ae3d317cc8d4020c86b5cb0e32f2a68b3a4e5d1e5338dfR191)
> 
> https://github.com/orientwalt/htdf/blob/097b2efd930335147e4454fbeb1a6a04d04f10e2/app/v0/protocol_v0.go#L191-L192

fixed at commit  a5b0cfa70ff38be0c04ad7374eec3f5d3e7bf5c3  , all test cases  in `test/pythontests` were passed.
